app/assets/javascripts/editorToolbar.js

```diff
@@ -90,9 +90,52 @@ $(function() {
     })
     // for save & recover buttons
     .on("click", ".save-button", function(e) {
-      $E.setState(e.currentTarget.dataset.formId); // string that is: "main", "reply-123", "edit-123" etc.
-      $E.save($E);
-    })
+      //preventing multiple clicks on the button
+      if(!e.detail || e.detail === 1) {
+          //explicitly handling main comment section
+          if ((this.id) === "save-button-main") {
+            // toggling the favicon save icon class to add a spinner icon
+            $(this).find("i").toggleClass("fa fa-save fas fa-sync fa-spin");
+      
+            //changing the text from "Upload an image" to "Saving..."
+            let saving_text = $('<p id="saving-text"> Saving... </p>');
+            $("#comment-form-main .imagebar").prepend(saving_text);
+            $("#comment-form-main .imagebar p").not("#saving-text").hide();
```

Comment:
Is this change of text to make it re-usable with the comment save? Meaning that if it stays `upload an image` it would mean that this text would appear when saving a text?

Reply:
Yes, indeed! The text `upload an image` would be hidden and in turn the text `Saving...` would be displayed to the user at the image upload text section of the comment form.

Reply:
The `upload an image` text is simply not interchanged with the `Saving...` text because the `upload an image` text also houses a hidden text, i.e. `Uploading` when an image is dragged on the section...

Reply:
Why not have a specific class or id for the text functionality and then hide the imagebar class/text? Is it easier to just reuse the imagebar class?

Reply:
That may involve refactoring text functionality class code / styles associated with the current imagebar id in different places...That's why imagebar class was reused 😅